File: code/java/matrix.java
package code.java;

import java.util.Random;

public class matrix {
    
    public static double[][] generateMatrix(int n) {
        double[][] matrix = new double[n][n];
        Random rand = new Random();
        for (int i = 0; i < n; i++) 
            for (int j = 0; j < n; j++) 
                matrix[i][j] = rand.nextDouble();
        return matrix;
    }

    public static double[][] multiplyMatrices(double[][] A, double[][] B, int n) {
        double[][] C = new double[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    C[i][j] += A[i][k] * B[k][j];
                }
            }
        }

        return C;
    }
}

File: code/c/matrix.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/time.h>

void initialize_matrix(double** matrix, int n) {
    for (int i = 0; i < n; ++i) 
        for (int j = 0; j < n; ++j) 
            matrix[i][j] = (double)rand() / RAND_MAX;
}

void multiply_matrices(double** a, double** b, double** c, int n) {
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            c[i][j] = 0.0;
            for (int k = 0; k < n; ++k) 
                c[i][j] += a[i][k] * b[k][j]; 
            
        }
    }
}
